prixcarburants/parse.py

```diff
@@ -28,48 +29,73 @@ def build_sale_points(filename: str) -> List[SalePoint]:
     return [SalePoint.build(element) for element in root]
 
 
-def _latest_metrics(latest_sale_points: List[dict]) -> dict:
-    """Build metrics about latest sale points"""
-    prices = {key.value: [0, 0] for key in FuelType}
-    for sale_point in latest_sale_points:
-        for key, (_, price) in sale_point["prices"].items():
-            prices[key][0] += price
-            prices[key][1] += 1
+def build_metrics(sale_points: List[dict]) -> dict:
+    """
+    Extract metrics from latest ``sale_points``
+    """
+    prices = {
+        department: {fuel_type.value: [0, 0] for fuel_type in FuelType}
+        for department in DEPARTMENTS
+    }
+    prices["total"] = {fuel_type.value: [0, 0] for fuel_type in FuelType}
+    for sale_point in sale_points:
+        for fuel_type, (_, price) in sale_point["prices"].items():
+            prices["total"][fuel_type][0] += price
+            prices["total"][fuel_type][1] += 1
+            department = int(sale_point["postcode"][:2])
+            prices[department][fuel_type][0] += price
+            prices[department][fuel_type][1] += 1
```

Comment:
Wouldn't it be better to compute the total at the end, like

```python
prices["total"] = {fuel_type.value: [sum(prices[:][fuel_type][0], sum(prices[:][fuel_type][1]] ...}
```

Reply:
Done https://github.com/Doreapp/prix-carburants/pull/36/commits/275ca6e61bcac3df78bfd8bb53ecf2d00d702b5c